index.tsx

```diff
@@ -42,43 +34,39 @@ const isTapInsideComponent = (target: any, nestedViewRef: any) => {
       }
     }
   }
+
+  return false;
 };
 
 /**
  * Wrapper component to detect event outside a specific element
  * Inherit from View, so it will take all View's props
  * @param OutsideViewProps - acceptance props
  */
-const OutsideView = ({
-  children,
+const OutsideView: FC<OutsideViewProps> = ({
   childRef,
   onPressOutside,
-  onMoveShouldSetResponder,
+  onStartShouldSetResponder,
   ...rest
-}: OutsideViewProps) => (
+}) => (
   <View
     {...rest}
     onStartShouldSetResponder={(evt: GestureResponderEvent) => {
       evt.persist();
 
       // if press outside, execute onPressOutside callback
       if (
+        onPressOutside &&
         childRef &&
         !isTapInsideComponent(evt.target, childRef.current || childRef)
       ) {
-        onPressOutside && onPressOutside();
-      }
-
-      // return onMoveShouldSetResponder in case it is passed to OutsideView
-      if (onMoveShouldSetResponder) {
-        return onMoveShouldSetResponder(evt);
```

Comment:
that strange to invoke `onMoveShouldSetResponder` inside a `onStartShouldSetResponder`